relay-dynamic-config/src/project.rs

```diff
@@ -19,6 +20,11 @@ use crate::metrics::{
 use crate::trusted_relay::TrustedRelayConfig;
 use crate::{GRADUATED_FEATURE_FLAGS, defaults};
 
+pub struct RetentionSettings {
+    pub standard: u16,
+    pub down_sampled: u16,
+}
+
```

Comment:
Rust beginner question: how do I implement the methods below? Is there a better option to use instead of `struct`?

```
error[E0277]: `RetentionSettings` doesn't implement `std::fmt::Debug`
error[E0277]: the trait bound `RetentionSettings: Clone` is not satisfied
error[E0277]: the trait bound `RetentionSettings: Serialize` is not satisfied
error[E0277]: the trait bound `RetentionSettings: Deserialize<'_>` is not satisfied
```

Reply:
You need to had the needed traits to your struct:
```rust
#[derive(Debug, Clone, Serialize, Deserialize)]
```